sql/core/src/test/scala/org/apache/spark/sql/InjectRuntimeFilterSuite.scala

```diff
@@ -207,7 +207,7 @@ class InjectRuntimeFilterSuite extends QueryTest with SQLTestUtils with SharedSp
 
     // `MergeScalarSubqueries` can duplicate subqueries in the optimized plan and would make testing
     // complicated.
-    conf.setConfString(SQLConf.OPTIMIZER_EXCLUDED_RULES.key, MergeScalarSubqueries.ruleName)
+    conf.setConfString(SQLConf.OPTIMIZER_EXCLUDED_RULES.key, MergeSubplans.ruleName)
```

Comment:
This also rings a bell to me. Is there a fallback plan for this improvement? For example, can we exclude this new implement without a regression which means keeping the existing `MergeScalarSubqueries`?